c_glib/gandiva-glib/expression.cpp

```diff
@@ -110,31 +111,23 @@ ggandiva_expression_class_init(GGandivaExpressionClass *klass)
 
 /**
  * ggandiva_expression_new:
- * @function: The function name in the expression.
- * @input_fields: (element-type GArrowField): The input fields.
- * @output_field: A #GArrowField to be output.
+ * @node: The node in the expression tree.
+ * @field: A #GArrowField to be output.
  *
  * Returns: (transfer full): The expression tree with a root node,
  *   and a result field.
  *
  * Since: 0.12.0
  */
 GGandivaExpression *
-ggandiva_expression_new(const gchar *function,
-                        GList *input_fields,
-                        GArrowField *output_field)
+ggandiva_expression_new(GGandivaNode *node,
+                        GArrowField *field)
 {
-  std::vector<std::shared_ptr<arrow::Field>> arrow_input_fields;
-  for (GList *node = input_fields; node; node = g_list_next(node)) {
-    auto field = GARROW_FIELD(node->data);
-    auto arrow_input_field = garrow_field_get_raw(field);
-    arrow_input_fields.push_back(arrow_input_field);
-  }
-  auto arrow_output_field = garrow_field_get_raw(output_field);
+  auto gandiva_node = ggandiva_node_get_raw(node);
+  auto arrow_field = garrow_field_get_raw(field);
   auto gandiva_expression =
-    gandiva::TreeExprBuilder::MakeExpression(function,
-                                             arrow_input_fields,
-                                             arrow_output_field);
+    gandiva::TreeExprBuilder::MakeExpression(gandiva_node,
+                                             arrow_field);
```

Comment:
We need to keep a reference to `GArrowField`.
See f36ad38538644110faad26ea8e963776880fe084 how to implement it.

Reply:
I've implemented it.

Reply:
And I’ve implemented to keep a reference to `GGandivaNode`.
Please review again.